lib/components/Table.js

```diff
@@ -0,0 +1,172 @@
+      {loading
+        ? Array(rowCount)
+          .fill()
+          .map((_, index) => (
+            <React.Fragment key={index}>
+              <tr>
+                {Array(columnCount)
+                  .fill()
+                  .map((_, index) => (
+                    <td className="text-center" key={index}>
+                      <div
+                        key={index}
+                        className={classnames(["skeleton-loader"])}
+                      >
+                          NA
+                      </div>
+                    </td>
+                  ))}
+              </tr>
+            </React.Fragment>
+          ))
+        : children}
+    </tbody>
+  );
+};
+
+Table.Foot = ({ children, ...rest }) => {
+  return <tfoot {...rest}>{children}</tfoot>;
+};
+
+Table.TH = ({
+  children = null,
+  isAscending = false,
+  isDescending = false,
+  ...rest
+}) => {
+  return (
+    <th
+      className={classnames({
+        "sort-ascending": isAscending,
+        "sort-descending": isDescending,
+      })}
+      {...rest}
+    >
+      {children}
+    </th>
+  );
+};
+
+Table.TR = ({ children }) => {
+  return <tr>{children}</tr>;
+};
+
+Table.TD = ({ className = null, center, children, ...rest }) => (
+  <td className={classnames({ "text-center": center }, [className])} {...rest}>
+    {children}
+  </td>
+);
+
+Table.Head.propTypes = {
+  children: PropTypes.oneOfType([
+    PropTypes.arrayOf(PropTypes.node),
+    PropTypes.node,
+  ]),
+};
+
+Table.Foot.propTypes = {
+  children: PropTypes.oneOfType([
+    PropTypes.arrayOf(PropTypes.node),
+    PropTypes.node,
+  ]),
+};
+
+Table.Body.propTypes = {
+  /**
+   * Display skeleton loading indicator
+   */
+  isLoading: PropTypes.shape({
+    loading: PropTypes.bool,
+    rowCount: PropTypes.number,
+    columnCount: PropTypes.number,
+  }),
+  children: PropTypes.oneOfType([
+    PropTypes.arrayOf(PropTypes.node),
+    PropTypes.node,
+  ]),
+};
+
+Table.TH.propTypes = {
+  isAscending: PropTypes.bool,
+  isDescending: PropTypes.bool,
+  className: PropTypes.string,
+  children: PropTypes.oneOfType([
+    PropTypes.arrayOf(PropTypes.node),
+    PropTypes.node,
+  ]),
+};
+
+Table.TR.propTypes = {
+  isLoading: PropTypes.object,
+  children: PropTypes.oneOfType([
+    PropTypes.arrayOf(PropTypes.node),
+    PropTypes.node,
+  ]),
+};
+
+Table.TD.propTypes = {
+  className: PropTypes.string,
+  children: PropTypes.oneOfType([
+    PropTypes.arrayOf(PropTypes.node),
+    PropTypes.node,
+  ]),
+};
+
+Table.propTypes = {
+  /**
+   * Update table styling to include checkbox
+   */
+  hasCheckbox: PropTypes.bool,
+  /**
+   * Update table styling to include actions
+   */
+  hasActions: PropTypes.bool,
+  children: PropTypes.oneOfType([
+    PropTypes.arrayOf(PropTypes.node),
+    PropTypes.node,
+  ]),
+};
+
+Table.Head.displayName = "Table.Head";
+Table.Body.displayName = "Table.Body";
+Table.Foot.displayName = "Table.Foot";
+Table.TH.displayName = "Table.TH";
```

Comment:
Can we rename to `Table.Th`, `Table.Tr`, `Table.Td`. If you can find a better name, then we can have that. Consistency is missing from the previous name.